zarr/indexing.py

```diff
@@ -825,3 +827,120 @@ def pop_fields(selection):
         selection = tuple(s for s in selection if not isinstance(s, str))
         selection = selection[0] if len(selection) == 1 else selection
     return fields, selection
+
+
+def int_to_slice(dim_selection):
+    return slice(dim_selection, dim_selection + 1, 1)
```

Comment:
Maybe we can just inline this since it is simple enough and used in only a couple places